web/pages/signin.js

```diff
@@ -43,112 +43,138 @@ export default function SignIn() {
+						}}
 					>
-						<AlertIcon />
-						<AlertDescription>{response.error}</AlertDescription>
-					</Alert>
-				)}
+						{({ errors, touched, isSubmitting, isValid }) => (
+							<Form>
+								<VStack justify="center" align="flex-start" spacing="0.5rem">
+									<CustomInputField
+										name="email"
+										label="Email"
+										error={errors.email}
+										touched={touched.email}
+										bg="white"
+									/>
+									<CustomInputField
+										name="password"
+										type="password"
+										label="Password"
+										error={errors.password}
+										touched={touched.password}
+										bg="white"
+									/>
+									<Box w="100%" align="center">
+										<Button
+											disabled={!isValid}
+											isLoading={isSubmitting}
+											mt="0.5rem"
+											colorScheme="red"
+											type="submit"
+										>
+											Sign In
+										</Button>
+									</Box>
+								</VStack>
+							</Form>
+						)}
+					</Formik>
 
-				<Formik
-					initialValues={initialValues}
-					validationSchema={validationSchema}
-					onSubmit={async (values, actions) => {
-						await onSignIn(values);
-						actions.setSubmitting(false);
-					}}
-				>
-					{({ errors, touched, isSubmitting, isValid }) => (
-						<Form>
-							<VStack justify="center" align="flex-start" spacing="0.5rem">
-								<CustomInputField
-									name="email"
-									label="Email"
-									error={errors.email}
-									touched={touched.email}
-									bg="white"
-								/>
-								<CustomInputField
-									name="password"
-									type="password"
-									label="Password"
-									error={errors.password}
-									touched={touched.password}
-									bg="white"
-								/>
-								<Box w="100%" align="center">
-									<Button
-										disabled={!isValid}
-										isLoading={isSubmitting}
-										mt="0.5rem"
-										colorScheme="red"
-										type="submit"
-									>
-										Sign In
-									</Button>
-								</Box>
-							</VStack>
-						</Form>
-					)}
-				</Formik>
+					<Flex justify="space-between" width="100%" direction="row">
+						<p>
+							<Link
+								style={{ textDecoration: 'inherit' }}
+								onClick={() => Router.push('/signup')}
+							>
+								Sign Up
+							</Link>
+						</p>
+						<p>
+							<Link style={{ textDecoration: 'inherit' }}>Forgot Password</Link>
+						</p>
+					</Flex>
+				</VStack>
+			</Flex>
+		);
+}
 
-				<Flex justify="space-between" width="100%" direction="row">
-					<p>
-						<Link
-							style={{ textDecoration: 'inherit' }}
-							onClick={() => Router.push('/signup')}
-						>
-							Sign Up
-						</Link>
-					</p>
-					<p>
-						<Link style={{ textDecoration: 'inherit' }}>Forgot Password</Link>
-					</p>
-				</Flex>
-			</VStack>
-		</Flex>
-	);
+export async function getServerSideProps({ req }) {
+	if (req.headers.cookie === null) return { props: { user: null } };
+	const user = await req.user;
```

Comment:
If this exists and we have access to it, do we even need to check if a cookie exists? Also I don't think req.user would be an async function, so we wouldn't need to await.